Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs

```diff
@@ -1006,6 +1007,55 @@ public void Cannot_Create_With_GeneralData_SubDataProcessor_If_UpdateMultiAssign
             AssertFailureWithKnownError(result, operationError, transaction);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Can_Create_With_GeneralData_MainContract(bool hasMainContract)
```

Comment:
more cases needed here when you change the implementation to use the app service